src/unstract/sdk/file_storage/permanent.py

```diff
@@ -28,10 +28,8 @@ def __init__(
                 f"supported in Permanent mode. "
                 f"Supported providers: {self.SUPPORTED_FILE_STORAGE_TYPES}"
             )
-        if provider == FileStorageProvider.GCS:
+        if provider == FileStorageProvider.GCS or provider == FileStorageProvider.LOCAL:
             super().__init__(provider, **storage_config)
-        elif provider == FileStorageProvider.LOCAL:
-            super().__init__(provider)
         else:
             raise NotImplementedError
```

Comment:
How this `NotImplementedError` diffr from `FileStorageError`. For exmple for Azure they will not get `FileStorageError` but `NotImplementedError`.

And `NotImplementedError` doesn't return what are we implemented or suported providers.

Reply:
NotImplemented is raised when we want to mean that though it is supported as a Permanent mode, we have not implemented it yet. Some conntributir can write the impl for Azure and then escape it from this error. However, FileStorageError in this context would mean that we do no intend to support if for PermanentFIleStorage at all.

Reply:
I have now added an error message also. Hope that helps.